pandas-stubs/io/sas/sasreader.pyi

```diff
@@ -1,12 +1,61 @@
-from typing import Sequence
-
-from pandas._typing import FilePathOrBuffer
-
-def read_sas(
-    path: FilePathOrBuffer,
-    format: str | None = ...,
-    index: Sequence | None = ...,
-    encoding: str | None = ...,
-    chunksize: int | None = ...,
-    iterator: bool = ...,
-): ...
+from abc import (
+    ABCMeta,
+    abstractmethod,
+)
+from typing import (
+    Hashable,
+    Literal,
+    overload,
+)
+
+from pandas import DataFrame
+
+from pandas._typing import (
+    CompressionOptions as CompressionOptions,
+    FilePath as FilePath,
+    ReadBuffer,
+)
+
+from pandas.io.sas.sas7bdat import SAS7BDATReader
+from pandas.io.sas.sas_xport import XportReader
+
+class ReaderBase(metaclass=ABCMeta):
+    @abstractmethod
+    def read(self, nrows: int | None = ...) -> DataFrame: ...
+    @abstractmethod
+    def close(self) -> None: ...
+    def __enter__(self) -> ReaderBase: ...
+    def __exit__(self, exc_type, exc_value, traceback) -> None: ...
+
+@overload
+def read_sas(
+    filepath_or_buffer: FilePath | ReadBuffer[bytes],
+    format: str | None = ...,
+    index: Hashable | None = ...,
+    encoding: str | None = ...,
+    *,
+    chunksize: int,
+    iterator: bool = ...,
+    compression: CompressionOptions = ...,
+) -> XportReader | SAS7BDATReader: ...

```

Comment:
If `format` is specified, we can then infer whether the result is `XportReader` or `SAS7BDATReader`.  This will double the number of overloads, but is probably more user friendly.  And if it is not specified, we could return `ReaderBase`, since presumably someone would use it in a context manager, and independent of the result, the methods needed to do reading would be present.